src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
+
+    if (!func_800EE29C(globalCtx, 0xC9)) {
+        this->actor.shape.rot.y = this->unk_258;
+    }
+}
+
+void func_8088EF18(Color_RGBAf* dest, Color_RGBAf* newColor, Color_RGBAf* curColor, f32 rate) {
+    Color_RGBAf rgbaDiff;
+
+    rgbaDiff.r = newColor->r - curColor->r;
+    rgbaDiff.g = newColor->g - curColor->g;
+    rgbaDiff.b = newColor->b - curColor->b;
+    rgbaDiff.a = newColor->a - curColor->a;
+
+    dest->r += rgbaDiff.r * rate;
+    dest->g += rgbaDiff.g * rate;
+    dest->b += rgbaDiff.b * rate;
+    dest->a += rgbaDiff.a * rate;
+}
+
+void func_8088EFA4(EnElf* this, GlobalContext* globalCtx) {
+    Actor* arrayPointerActor = globalCtx->actorCtx.targetContext.unk38;
+    Player* player = GET_PLAYER(globalCtx);
+    f32 transitionRate;
+
+    if (!(this->unk_264 & 8)) {
+        this->unk_264 &= ~0x10;
+    }
+
+    if (this->unk_264 & 8) {
+        if (this->unk_264 & 0x10) {
+            this->unk_238 = 1.0f;
+            this->unk_264 &= ~8;
+            this->unk_268 = 0;
+            this->unk_264 &= ~0x10;
+        } else {
+            this->innerColor.r = 255.0f;
+            this->innerColor.g = 0.0f;
+            this->innerColor.b = 0.0f;
+            this->innerColor.a = 255.0f;
+
+            this->outerColor.r = 200.0f;
+            this->outerColor.g = 80.0f;
+            this->outerColor.b = 80.0f;
+            this->outerColor.a = 0.0f;
+        }
+    } else if (globalCtx->actorCtx.targetContext.unk40 != 0.0f) {
+        this->unk_268 = 0;
+        this->unk_238 = 1.0f;
+        if (!this->unk_269) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_BELL_DASH_NORMAL);
+        }
+    } else if (this->unk_268 == 0) {
+        if ((arrayPointerActor == NULL) ||
+            (Math_Vec3f_DistXYZ(&this->actor.world.pos, &globalCtx->actorCtx.targetContext.unk0) < 50.0f)) {
+            this->unk_268 = 1;
+        }
+    } else if (this->unk_238 != 0.0f) {
+        if (Math_StepToF(&this->unk_238, 0.0f, 0.25f)) {
+            this->innerColor = globalCtx->actorCtx.targetContext.unk18;
+            this->outerColor = globalCtx->actorCtx.targetContext.unk28;
+        } else {
+            transitionRate = 0.25f / this->unk_238;
+            func_8088EF18(&this->innerColor, &globalCtx->actorCtx.targetContext.unk18, &this->innerColor,
+                          transitionRate);
+            func_8088EF18(&this->outerColor, &globalCtx->actorCtx.targetContext.unk28, &this->outerColor,
+                          transitionRate);
+        }
+    }
+
+    if (this->fairyFlags & 1) {
+        if ((arrayPointerActor == NULL) || (player->unk_730 == NULL)) {
+            this->fairyFlags ^= 1;
+        }
+    } else if ((arrayPointerActor != NULL) && (player->unk_730 != NULL)) {
+        u8 temp = this->unk_269;
+        u16 targetSound = this->unk_269 == 0 ? NA_SE_PL_WALK_GROUND - SFX_FLAG : NA_SE_PL_WALK_GROUND - SFX_FLAG;
+
+        if (!temp) {
+            Audio_PlayActorSound2(&this->actor, targetSound);
+        }
```

Comment:
```suggestion
        u16 targetSfxId = this->unk_269 == 0 ? NA_SE_PL_WALK_GROUND - SFX_FLAG : NA_SE_PL_WALK_GROUND - SFX_FLAG;

        if (!temp) {
            Audio_PlayActorSound2(&this->actor, targetSfxId);
        }
```
`SfxId` is the name we use for sound effects